grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -183,7 +183,12 @@
         requestFramer = new ArmeriaMessageFramer(ctx.alloc(), maxOutboundMessageSizeBytes, grpcWebText);
         marshaller = new GrpcMessageMarshaller<>(ctx.alloc(), serializationFormat, method, jsonMarshaller,
                                                  unsafeWrapResponseBuffers);
-        executor = callOptions.getExecutor();
+
+        if (callOptions.getExecutor() == null) {
+            contextAwareExecutor = ctx.makeContextAware(MoreExecutors.directExecutor());
+        } else {
+            contextAwareExecutor = ctx.makeContextAware(callOptions.getExecutor());
```

Comment:
I wonder if we should wrap user-defined `callOptions.getExecutor()` with sequential executor to ensure linearization of events. Any opinions?

Reply:
They do in fact [wrap it with serializing executor](https://github.com/grpc/grpc-java/blob/4e3ee4471e671a67e012cd1a90280c988f88147a/core/src/main/java/io/grpc/internal/ClientCallImpl.java#L109-L118). Let's do it too.